keps/sig-storage/3751-volume-attributes-class/README.md

```diff
@@ -779,6 +789,9 @@ These goals will help you determine what you need to measure (SLIs) in the next
 question.
 -->
 
+- Ratio of `controller_modify_volume_errors_total`/`controller_modify_volume_total` <= 1%
```

Comment:
Many providers have a rate limit on how often you can modify the volume. In our metrics, do we need to explicitly call out some error code to indicate that quota has been exhausted?

Reply:
Shall we add `RESOURCE_EXHAUSTED`  to [CSI Spec: ControllerModifyVolume errors](https://github.com/container-storage-interface/spec/blob/master/spec.md#controllermodifyvolume-errors)? 

Today EBS CSI Driver treats hitting rate-limits or our 6 hour modification cooldown as `INTERNAL`, which is not ideal metrics-wise. `RESOURCE_EXHAUSTED` would be an improvement here. 

Thoughts @msau42?

Reply:
gnufied@ proposed `UNAVAILABLE` instead for rate-limit related errors. I agree that this makes sense for the transient rate-limit errors (e.g. where request tokens are refilled every second if rate-limiting is using token-bucket algorithm).

```
The service is currently unavailable. This is most likely a transient condition, which can be corrected by retrying with a backoff. Note that it is not always safe to retry non-idempotent operations.
```

For longer cooldown style rate-limits like EBS 6 hour timer between modifications of a volume, I'm not sure if this error code makes sense because it is not a final error that would lead to `infeasible` request status. No infeasible request status means that the sidecar would keep retrying this RPC despite the cooldown which leads to extra calls to the cloud provider. However other error codes like `FAILED_PRECONDITION` are already in use in `ControllerExpandVolume` so I'm not sure that solving this cooldown case Cloud Provider API spam is solvable today.

Reply:
UNAVAILABLE makes sense to me. Since it's not a final error though, I think any throttling may need to be handled in the CSI driver.